pgxn/neon/pagestore_smgr.c

```diff
@@ -145,6 +145,25 @@ static UnloggedBuildPhase unlogged_build_phase = UNLOGGED_BUILD_NOT_IN_PROGRESS;
 /* Max amount of tracked buffer reads */
 #define READ_BUFFER_SIZE 128
```

Comment:
Then you should change your prefetch algorithm to handle values >= 128 correctly (just use Min(READ_BUFFER_SIZE, seqscan_prefetch_buffers))
Because right now them just switch of prefetch. 
It is not what user expected. This value (128) is completely hidden from end user, so he will be very surprsed
why with prefetch_buffers=100 query runs fast and with prefetch_buffers=200  several times slower.